physionet-django/project/migrations/0050_parse_legacy.py

```diff
@@ -0,0 +1,24 @@
+from django.db import migrations, models
+import django.db.models.deletion
+
+from project.models import PublishedProject
+
+def parse_legacy(apps, schema_editor):
+    projects = PublishedProject.objects.filter(is_legacy=True)
+    for p in projects:
+        p.parse_legacy_content()
+
+def unparse_legacy(apps, schema_editor):
+    projects = PublishedProject.objects.filter(is_legacy=True)
+    for p in projects:
+        p.project_content.all().delete()
```

Comment:
This for loop leads to the famous N+1 queries problem. When you access a related model, a separate query is executed. In order to improve it, you should add `prefetch_related('project_content')` to the initial queryset.

You can also achieve that in one line:
 ```suggestion
def unparse_legacy(apps, schema_editor):
    PublishedSectionContent.objects.filter(project__is_legacy=True).delete()
```

Reply:
Done.